src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -0,0 +1,239 @@
+            }
+            else
+            {
+                _searchEventHandler(*this, _textBox.Text());
+            }
+        }
+    }
+
+    // Method Description:
+    // - Handler for pressing Enter on TextBox, trigger
+    //   text search
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::SetFocusOnTextbox()
+    {
+        if (_textBox)
+        {
+            Input::FocusManager::TryFocusAsync(_textBox, FocusState::Keyboard);
+            _textBox.SelectAll();
+        }
+    }
+
+    // Method Description:
+    // - Check if the current focus is on any element within the
+    //   search box
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - bool: whether the current focus is on the search box
+    bool SearchBoxControl::ContainsFocus()
+    {
+        auto focusedElement = Input::FocusManager::GetFocusedElement(this->XamlRoot());
+        if (_focusableElements.count(focusedElement) > 0)
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    // Method Description:
+    // - Handler for clicking the GoBackward button. This change the value of _goForward,
+    //   mark GoBackward button as checked and ensure GoForward button
+    //   is not checked
+    // Arguments:
+    // - sender: not used
+    // - e: not used
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::_GoBackwardClicked(winrt::Windows::Foundation::IInspectable const& /*sender*/, RoutedEventArgs const& /*e*/)
+    {
+        _goForward = false;
+        _goBackwardButton.IsChecked(true);
+        if (_goForwardButton.IsChecked())
+        {
+            _goForwardButton.IsChecked(false);
+        }
+
+        // kick off search
+        _searchEventHandler(*this, _textBox.Text());
+    }
+
+    // Method Description:
+    // - Handler for clicking the GoForward button. This change the value of _goForward,
+    //   mark GoForward button as checked and ensure GoBackward button
+    //   is not checked
+    // Arguments:
+    // - sender: not used
+    // - e: not used
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::_GoForwardClicked(winrt::Windows::Foundation::IInspectable const& /*sender*/, RoutedEventArgs const& /*e*/)
+    {
+        _goForward = true;
+        _goForwardButton.IsChecked(true);
+        if (_goBackwardButton.IsChecked())
+        {
+            _goBackwardButton.IsChecked(false);
+        }
+
+        // kick off search
+        _searchEventHandler(*this, _textBox.Text());
+    }
+
+    // Method Description:
+    // - Handler for clicking the Case Sensitivity button. This changes
+    //   _isCaseSensitive value
+    // Arguments:
+    // - sender: not used
+    // - e: not used
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::_CaseSensitivityClicked(winrt::Windows::Foundation::IInspectable const& /*sender*/, RoutedEventArgs const& /*e*/)
+    {
+        _isCaseSensitive = !_isCaseSensitive;

```

Comment:
We want to make sure that the state of the button never gets de-synced from the state of this boolean.
Do we need the boolean? Or can we query the button's current state?